### ISSUE TITLE
Linux non-root install failure: Create BOND_ENABLE_BASHCOMPLETION

### ISSUE THREAD

**Issue by SoyYoRafa** [Microsoft/bond#932]
Hi,
I would like to install Bond in an environment where I do not have root access. It fails because it tries to create /etc/bash_completion.d/gbc. The library archive happens to be installed afterwards so it is important to install the remaining files. An easy fix would be to set BOND_ENABLE_BASHCOMPLETION=FALSE in cmake, similar to gRPC. 

CMake Error at compiler/cmake_install.cmake:53 (file):
  file INSTALL cannot copy file
  "/home/me/bond_src/build/compiler/gbc.comp" to
  "/etc/bash_completion.d/gbc".
Call Stack (most recent call first):
  cmake_install.cmake:71 (include)

**Comment by lalo** [comment]
Sounds like you have some knowledge, care to submit a PR?

Also before working on a PR, play around with this cmake var https://cmake.org/cmake/help/v3.0/variable/CMAKE_INSTALL_PREFIX.html

`On UNIX one can use the DESTDIR mechanism in order to relocate the whole installation. DESTDIR means DESTination DIRectory. It is commonly used by makefile users in order to install software at non-default location. It is usually invoked like this: make DESTDIR=/home/john install`

It might help if you try installing to a folder were you have write permissions.

**Comment by SoyYoRafa** [comment]
Happy to submit a PR, just following contribution guidelines that request that I post here first. 
The destination is hard coded in compiler/CMakeLists.txt. Presumably because the author wanted it to be picked up automatically by bash. As a result, using CMAKE_INSTALL_PREFIX isn't going to give equivalent behavior. I can try checking write permissions but there isn't a native way in cmake. I can try using execute_process, I would have to play around with it.

`set (completion_dir /etc/bash_completion.d)`

```
set (completion_dir /etc/bash_completion.d)
...
if(EXISTS "${completion_dir}" AND IS_DIRECTORY "${completion_dir}")
    install (FILES ${completion}
        RENAME gbc
        DESTINATION ${completion_dir})
endif()
```

**Comment by lalo** [comment]
Closed with f78a418
